src/components/withNavigationFocus.tsx

```diff
@@ -0,0 +1,24 @@
+import React, {ComponentType, ForwardedRef, RefAttributes} from 'react';
+import {useIsFocused} from '@react-navigation/native';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+type WithNavigationFocusProps = {
+    isFocused: boolean;
+};
+
+export default function withNavigationFocus<TProps extends WithNavigationFocusProps, TRef>(WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>) {
+    function WithNavigationFocus(props: Omit<TProps, keyof WithNavigationFocusProps>, ref: ForwardedRef<TRef>) {
```

Comment:
lets add return type

Reply:
You mean to add return type to `withNavigationFocus` function, right? @kubabutkiewicz

Reply:
right :D